cmake/app/boilerplate.cmake

```diff
@@ -20,7 +20,12 @@
 # policy is used, but we need policy CMP0000 set to OLD to avoid
 # copy-pasting cmake_minimum_required across application
 # CMakeLists.txt files.
-cmake_minimum_required(VERSION 3.8.2)
+#
+# 3.6.0 is set as the minimum required version because MSYS does not
+# support 3.8, this will be set back to 3.8 when
+# https://github.com/zephyrproject-rtos/zephyr/issues/4687 is
+# resolved.
+cmake_minimum_required(VERSION 3.6.0)
```

Comment:
Why not do the uname first, and then if(MSYS) 3.6 else 3.8?

Reply:
Because we document what the supported CMake version is. So we don't need CMake to enforce the version for us.

We could have cmake_minimum_required(1.0.0)
here if we wanted.

Reply:
I still don't see what's wrong with failing early on Linux/macOS if cmake < 3.8, even if we document it